src/config.c

```diff
@@ -225,8 +229,33 @@ bool load_configuration(const char *override_configpath, config_load_t load_type
             "$XDG_CONFIG_HOME/i3/config, ~/.i3/config, $XDG_CONFIG_DIRS/i3/config "
             "and " SYSCONFDIR "/i3/config)");
     }
-    LOG("Parsing configfile %s\n", current_configpath);
-    const bool result = parse_file(current_configpath, load_type != C_VALIDATE);
+
+    IncludedFile *file;
+    while (!TAILQ_EMPTY(&included_files)) {
+        file = TAILQ_FIRST(&included_files);
+        FREE(file->path);
+        TAILQ_REMOVE(&included_files, file, files);
+        FREE(file);
+    }
+
+    char resolved_path[PATH_MAX] = {'\0'};
+    if (realpath(current_configpath, resolved_path) == NULL) {
+        die("realpath(%s): %s", current_configpath, strerror(errno));
+    }
```

Comment:
It seems that the root file can be included by an `include` directive. This seems to fix that:

```suggestion
    }
    
    file = scalloc(1, sizeof(IncludedFile));
    file->path = sstrdup(resolved_path);
    TAILQ_INSERT_TAIL(&included_files, file, files);
    
```

Reply:
Good catch, done!